R/FilteredData.R

```diff
@@ -875,7 +881,19 @@ FilteredData <- R6::R6Class( # nolint
 
             header_html <- tags$tr(
               tagList(
-                lapply(colnames(datasets_df), tags$td)
+                lapply(
+                  vapply(
+                    seq_along(datasets_df),
+                    function(i) {
+                      ifelse(
+                        !is.null(attr(datasets_df[[i]], "label")),
+                        attr(datasets_df[[i]], "label"),
+                        names(datasets_df)[[i]]
```

Comment:
Anyway you can substitute `names(datasets_df)[[i]]` with `names(datasets_df[[i]])` ? Then you can assign

`ds <- datasets_df[[i]]` and have shorter usage of this part later in the code when taking `names(ds)` or `attr(ds)`?

Reply:
thanks, all valid points.

Let me reorg this a little. I was initially doing this from the webbrowser - never again...

Reply:
actually this comment is incorrect. I cannot do `names(df[[i]])`

```
r$> i <- 1

r$> names(iris)[[1]]
[1] "Sepal.Length"

r$> names(iris[[1]])
NULL
```

Reply:
this is the main reason for looping through the integers - to be able to call `names(df)[[i]]`

Reply:
I wasn't sure about that. I think for the names it would need to be `iris[1]`. So ignore this one